app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.order completion_date: :desc, deadline: :asc
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id].to_i)
```

Comment:
Also good to verify that you found the Task, if you didn't then redirect them someplace.  That's a good defensive practice.